File: CS/Form1.cs
using DevExpress.XtraRichEdit.API.Native;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace SubDocumentIterator
{
    public partial class Form1 : DevExpress.XtraEditors.XtraForm
    {
        public Form1()
        {
            InitializeComponent();
            richEditControl1.LoadDocument("template.docx");
        }

        private void updateFieldsBtn_ItemClick(object sender, DevExpress.XtraBars.ItemClickEventArgs e)
        {
            richEditControl1.Document.ForEachSubDocument((subdoc => { subdoc.Fields.Update(); }));
        }

        private void removeBookmarksBtn_ItemClick(object sender, DevExpress.XtraBars.ItemClickEventArgs e)
        {
            richEditControl1.Document.ForEachSubDocument((subdoc => 
            {
                for (int i = subdoc.Bookmarks.Count - 1; i >= 0; i--)
                    subdoc.Bookmarks.Remove(subdoc.Bookmarks[i]);
            }));
        }

        private void replaceTextBtn_ItemClick(object sender, DevExpress.XtraBars.ItemClickEventArgs e)
        {
            richEditControl1.Document.ForEachSubDocument((subdoc => { subdoc.ReplaceAll("test text", "Hello!!!", SearchOptions.None); }));
        }

        private void highlightBtn_ItemClick(object sender, DevExpress.XtraBars.ItemClickEventArgs e)
        {
            richEditControl1.Document.ForEachSubDocument((subdoc =>
            {
                DocumentRange[] ranges = subdoc.FindAll("time", SearchOptions.None);
                foreach(DocumentRange range in ranges)
                {
                    CharacterProperties cp = subdoc.BeginUpdateCharacters(range);
                    cp.ForeColor = Color.Red;
                    cp.BackColor = Color.Lavender;
                    subdoc.EndUpdateCharacters(cp);
                }
            }));
        }
    }
}

File: VB/Form1.vb
Imports DevExpress.XtraRichEdit.API.Native
Imports System.ComponentModel
Imports System.Drawing

Namespace SubDocumentIterator

    Public Partial Class Form1
        Inherits DevExpress.XtraEditors.XtraForm

        Public Sub New()
            InitializeComponent()
            richEditControl1.LoadDocument("template.docx")
        End Sub

        Private Sub updateFieldsBtn_ItemClick(ByVal sender As Object, ByVal e As DevExpress.XtraBars.ItemClickEventArgs)
            richEditControl1.Document.ForEachSubDocument((Sub(subdoc) subdoc.Fields.Update()))
        End Sub

        Private Sub removeBookmarksBtn_ItemClick(ByVal sender As Object, ByVal e As DevExpress.XtraBars.ItemClickEventArgs)
            richEditControl1.Document.ForEachSubDocument(Sub(subdoc)
                For i As Integer = subdoc.Bookmarks.Count - 1 To 0 Step -1
                    subdoc.Bookmarks.Remove(subdoc.Bookmarks(i))
                Next
            End Sub)
        End Sub

        Private Sub replaceTextBtn_ItemClick(ByVal sender As Object, ByVal e As DevExpress.XtraBars.ItemClickEventArgs)
            richEditControl1.Document.ForEachSubDocument(Sub(subdoc) subdoc.ReplaceAll("test text", "Hello!!!", SearchOptions.None))
        End Sub

        Private Sub highlightBtn_ItemClick(ByVal sender As Object, ByVal e As DevExpress.XtraBars.ItemClickEventArgs)
            richEditControl1.Document.ForEachSubDocument((Sub(subdoc)
                Dim ranges As DocumentRange() = subdoc.FindAll("time", SearchOptions.None)
                For Each range As DocumentRange In ranges
                    Dim cp As CharacterProperties = subdoc.BeginUpdateCharacters(range)
                    cp.ForeColor = Color.Red
                    cp.BackColor = Color.Lavender
                    subdoc.EndUpdateCharacters(cp)
                Next
            End Sub))
        End Sub
    End Class
End Namespace
